packages/arb-avm-cpp/avm_values/include/avm_values/codepointstub.hpp

```diff
@@ -20,14 +20,17 @@
 #include <avm_values/bigint.hpp>
 #include <avm_values/opcodes.hpp>
 
+#include <cassert>
+
 struct CodePoint;
 
 struct CodePointRef {
     uint64_t segment;
     uint64_t pc;
 
-    CodePointRef(uint64_t segment_, uint64_t pc_)
-        : segment(segment_), pc(pc_) {}
+    CodePointRef(uint64_t segment_, uint64_t pc_) : segment(segment_), pc(pc_) {
+        assert(segment < (uint64_t(1) << 62));
```

Comment:
Should this be more than just an assert?

Reply:
There's an exception based check in code.hpp addSegment. I figure that happens a lot less, and we can afford a more expensive check there, and just an assert here.